mopidy_mpd/network.py

```diff
@@ -122,19 +130,21 @@ def __init__(
         self.watcher = self.register_server_socket(self.server_socket.fileno())
 
     def create_server_socket(self, host, port):
-        socket_path = get_unix_socket_path(host)
-        if socket_path is not None:  # host is a path so use unix socket
-            sock = create_unix_socket()
-            sock.bind(socket_path)
-        else:
-            # ensure the port is supplied
-            if not isinstance(port, int):
-                raise TypeError(f"Expected an integer, not {port!r}")
-            sock = create_tcp_socket()
-            sock.bind((host, port))
-
-        sock.setblocking(False)
-        sock.listen(1)
+        sock = get_systemd_socket()
+        if sock is None:
+            socket_path = get_unix_socket_path(host)
+            if socket_path is not None:  # host is a path so use unix socket
+                sock = create_unix_socket()
+                sock.bind(socket_path)
+            else:
+                # ensure the port is supplied
+                if not isinstance(port, int):
+                    raise TypeError(f"Expected an integer, not {port!r}")
+                sock = create_tcp_socket()
+                sock.bind((host, port))
+
+            sock.setblocking(False)
+            sock.listen(1)
```

Comment:
Ah, good catch. We should probably not call `.setblocking(False)` and `.listen(1)` on the socket we get from systemd? I assume it is already ready for use?

I'd flip the first if so that we return early if we already have a socket and avoid indenting the rest of the function:

```suggestion
        sock = get_systemd_socket()
        if sock is not None:
            return sock

        socket_path = get_unix_socket_path(host)
        if socket_path is not None:  # host is a path so use unix socket
            sock = create_unix_socket()
            sock.bind(socket_path)
        else:
            if not isinstance(port, int):
                raise TypeError(f"Expected an integer, not {port!r}")
            sock = create_tcp_socket()
            sock.bind((host, port))

        sock.setblocking(False)
        sock.listen(1)
```